main.tf

```diff
@@ -1,152 +1,165 @@
-  subnet_id    = module.network.subnet_ids["workers"]
-  subnet_label = module.network.worker_subnet_dns_label
-
-  # worker compute parameters
-  worker_image_id     = var.worker_image_id
-  worker_shape        = var.worker_shape
-  worker_size         = var.worker_size
-  worker_upgrade      = var.worker_package_upgrade
-  ssh_public_key      = ""
-  ssh_public_key_path = var.ssh_public_key_path
-  timezone            = var.worker_timezone
-
-  # olcne public load balancer to attach to worker pool
-  oci_loadbalancer_id = module.loadbalancer.pub_lb_id
+module "infrastructure" {
+  depends_on = [null_resource.variable_validation]
+  source     = "./modules/terraform-oci-ocne-infrastructure"
+
+  availability_domain_id   = var.availability_domain_id
+  compartment_id           = var.compartment_id
+  prefix                   = var.prefix
+  subnet_id                = var.deploy_networking ? module.oci-ocne-network[0].private_subnet_id.*.id[0] : var.subnet_id
+  instance_shape           = var.instance_shape
+  image_ocid               = var.image_ocid
+  os_version               = var.os_version
+  kernel_version           = var.kernel_version
+  load_balancer_shape      = var.load_balancer_shape
+  ssh_public_key_path      = var.ssh_public_key_path
+  ssh_private_key_path     = var.ssh_private_key_path
+  control_plane_node_count = var.control_plane_node_count
+  worker_node_count        = var.worker_node_count
+  standalone_api_server    = var.standalone_api_server
+  yum_repo_url             = var.yum_repo_url
+  ocne_version             = data.external.ocne_config.result.version
+  enable_bastion          = var.enable_bastion
+  bastion_public_ip        = var.enable_bastion ? module.oci-ocne-network[0].bastion_public_ip : var.bastion_public_ip
+  bastion_user             = var.bastion_user
+  bastion_private_key_path = var.enable_bastion || var.bastion_public_ip != "" ? var.bastion_private_key_path : ""
+  compute_user             = var.compute_user
+  freeform_tags            = var.freeform_tags
+  virtual_ip               = var.virtual_ip
+  boot_volume_size_in_gbs  = var.boot_volume_size_in_gbs
 }
 
-module "loadbalancer" {
-  source = "./modules/loadbalancer"
-
-  # general parameters
-  compartment_id = var.compartment_id
-
-  label_prefix = var.label_prefix
-
-  nsg_ids = module.network.nsg_ids
-
-  # load balancer shapes
-  int_lb_shape = var.int_lb_shape
-  pub_lb_shape = var.public_lb_shape
-
-  # subnet ids
-  int_lb_subnet_id = lookup(module.network.subnet_ids, "int_lb")
-  pub_lb_subnet_id = lookup(module.network.subnet_ids, "pub_lb")
-
-  # workers ip addresses for backend resources
-  olcne_workers = module.worker.worker_ip_list
+module "vault" {
+  source     = "./modules/terraform-oci-vault"
+  count      = var.use_vault ? 1 : 0
+
+  user_id                  = var.user_id
+  region                   = var.region
+  tenancy_id               = var.tenancy_id
+  fingerprint              = var.fingerprint
```

Comment:
Why is the provider parameters being passed to the submodule here? If you need to initialize another provider you can do that in the root itself and pass a list of providers. see a more elaborate example here: https://github.com/oracle-terraform-modules/terraform-oci-verrazzano/blob/main/main.tf#L36

Doing so will reduce the number of variables you need to pass between modules